scripts/influxd-systemd-start.sh

```diff
@@ -1,5 +1,51 @@
 #!/bin/bash -e
 
+# Retrieve configuration value from influxd.
+function influxd_config() {
+    local header="${1}"
+    local target="${2}"
+
+    while IFS= read -r line
+    do
+        # The TOML specification allows for key-value pairs to be namespaced in
+        # collections. Therefore, it is not enough to grep for the desired key.
+        # This filters out all values that are outside the target section.
+        if [[ "${line}" =~ ^\[${header}\]$ ]]
+        then
+            local section_found=1
+        fi
+
+        # blank line signifies that the section has ended
+        if [[ ! ${line} ]]
+        then
+            local section_found=
+        fi
+
+        if [[ ${section_found:-} ]]
+        then
+                # Once within the target section, search the key-value pairs for the
+                # desired key. Currently, this only supports string values. Since
+                # this is used only for wal-dir and data-dir, this should be okay.
+                if [[ "${line}" =~ ^[[:space:]]*${target}[[:space:]]*=[[:space:]]\"(.*)\" ]]
+                then
+                    echo "${BASH_REMATCH[1]}" ; return
+                fi
+        fi
+    done <<< "$(influxd config -config /etc/influxdb/influxdb.conf 2>/dev/null)"
```

Comment:
Should `/etc/influxdb/influxdb.conf` be a constant/parameter that we use here and in the actual command line, so they can't get out of sync?

Reply:
The config is hard-coded on `master-1.x`: [`scripts/influxd-systemd-start.sh`](https://github.com/influxdata/influxdb/blob/53856cdaae63a699946b388ffbba5bb56784ba51/scripts/influxd-systemd-start.sh#L3). I am unsure what supplying another `-config <path>` would do (override vs ignore vs error), but the behavior should be consistent with the call that starts the service.

Reply:
I tested it, if `-config` is supplied via `INFLUXD_OPTS`, the first will be overridden in favor of `INFLUXD_OPTS`.